GenericJSON/Initialization.swift

```diff
@@ -96,3 +98,21 @@ extension JSON: ExpressibleByStringLiteral {
         self = .string(value)
     }
 }
+
+// MARK: - NSNumber
+
+extension NSNumber {
+    fileprivate var isBool: Bool {
```

Comment:
This is slightly magical. In such cases I like to provide a link to an extended discussion of the problem, to have something to hold onto when someone (likely us!) reads the code in future. Maybe [the Stack Overflow question](https://stackoverflow.com/questions/49641102) you have linked to in the issue? Or even just a link to the issue on GitHub.

Reply:
Yes, it is a bit magical. `NSNumber` appears to be an involved topic. There is a surprising amount of debate around how best to test if an `NSNumber` is a Bool.

I've added docs and a link to the relevant stack overflow question – should we need to revisit it.